source/core/slang-offset-container.cpp

```diff
@@ -72,62 +69,64 @@ size_t RelativeString::calcEncodedSize(size_t size, uint8_t encode[kMaxSizeEncod
     }
 }
 
-/* static */size_t RelativeString::calcAllocationSize(size_t stringSize)
+/* static */size_t OffsetString::calcAllocationSize(size_t stringSize)
 {
     uint8_t encode[kMaxSizeEncodeSize];
     size_t encodeSize = calcEncodedSize(stringSize, encode);
     // Add 1 for terminating 0
     return encodeSize + stringSize + 1;
 }
 
-/* static */size_t RelativeString::calcAllocationSize(const UnownedStringSlice& slice)
+/* static */size_t OffsetString::calcAllocationSize(const UnownedStringSlice& slice)
 {
     return calcAllocationSize(slice.size());
 }
 
-UnownedStringSlice RelativeString::getSlice() const
+UnownedStringSlice OffsetString::getSlice() const
 {
     size_t size;
     const char* chars = decodeSize(m_sizeThenContents, size);
 
     return UnownedStringSlice(chars, size);
 }
 
-const char* RelativeString::getCstr() const
+const char* OffsetString::getCstr() const
 {
     return getSlice().begin();
 }
 
-/* !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! RelativeContainer !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! */
+/* !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! OffsetContainer !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! */
 
-RelativeContainer::RelativeContainer()
+OffsetContainer::OffsetContainer()
 {
-    m_current = 0;
     m_capacity = 0;
-    m_base.m_data = nullptr;
+    m_data = nullptr;
+
+    // We need to allocate some of the first bytes 0 can be used for nullptr. 
```

Comment:
I assume you mean "some of the first bytes *so that* 0 can be used for nullptr"?